### PR TITLE
Update aws-kinesis-gluejob construct to allocate a capacity of 2 in the default props (instead of 100).

### PR DESCRIPTION
**Pull request by knihit**
*Issue #, if available:*

*Description of changes:*
The CfnJobProps, default property did not set the maximum DPUs to use for streaming ETL jobs. This meant that the Glue service would create upto 100 DPUs. The default props now explicitly set the capacity to the minimum available value which is 2. When using the construct, you can always override the default configuration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 1ab1c91a00b8aa96e7b6bd6b641d3cfb29a9dfc9
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0813367f-125f-4f46-920c-56797b7284a0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/lib/glue-job-defaults.ts [hunk 1]
```diff
@@ -11,20 +11,35 @@
...
+
+  let defaultGlueJobProps: glue.CfnJobProps;
+
+  if (userProvidedGlueJobProps.workerType !== undefined && userProvidedGlueJobProps.numberOfWorkers !== undefined) {
+    defaultGlueJobProps = {
+      command: userProvidedGlueJobProps.command!,
+      role: jobRole.roleArn,
+      securityConfiguration: glueSecurityConfigName,
+      defaultArguments: _defaultArguments,
+      maxCapacity: 2, // setting default to 2 to ensure customers don't pay for something they don't need
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/glue-job-defaults.ts:29]: When I read this code it appears that if the user specifies workerType and numberOfWorkers, then we set maxCapacity to 2 even if the user provided one. There does not appear to be any place where a client specified maxCapacity is recognized - nor any other userProvidedGlueJobProps attributes.

Regular pattern is the service has a set of default props. When users provide props, then overrideProps is used to replace the default props with the user provided values. When there are values that _must_ have a certain value for the construct to work, then overrideProps is run a second time to override the user provided values.

We may need to have a call to discuss.

File: source/patterns/@aws-solutions-constructs/core/test/glue-job-helper.test.ts [hunk 2]
```diff
@@ -32,11 +32,15 @@ test('Test deployment with role creation', () => {
     assumedBy: new ServicePrincipal('glue.amazonaws.com')
   });
 
-  const cfnJobProps: CfnJobProps = defaults.DefaultGlueJobProps(_jobRole, {
-    name: _jobID,
-    pythonVersion: '3',
-    scriptLocation: 's3://fakelocation/script'
-  }, 'testETLJob', {}, '1.0');
+  const _userProvidedCfnJobProps: Partial<CfnJobProps> = {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/test/glue-job-helper.test.ts:35]: The Solutions Constructs team is _considering_ moving to this to replace `| any`. We shouldn't be rolling it out all through the library at this point.

**Comment by knihit** [comment]
Replacing it with another one.

**Pull request closed without merging**